src/System.Net.Http/tests/UnitTests/HPack/HPackRoundtripTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http.HPack;
+using System.Net.Http.Headers;
+using System.Text;
+using Xunit;
+
+namespace System.Net.Http.Unit.Tests.HPack
+{
+    public class HPackRoundtripTests
+    {
+
+        public static IEnumerable<object[]> TestHeaders =>
+            new HttpHeaders[]
+            {
+                BuildHttpHeaders(("header", new[] { "value" })),
+
+                BuildHttpHeaders(("header", new[] { "value1", "value2" })),
+
+                BuildHttpHeaders(
+                    ("header-0", new[] { "value1", "value2" }),
+                    ("header-0", new[] { "value3" }),
+                    ("header-1", new[] { "value1" }),
+                    ("header-2", new[] { "value1", "value2" }))
+
+            }.Select(h => new[] { h });
+
+        [Theory, MemberData(nameof(TestHeaders))]
+        public void HPack_HeaderEncodeDecodeRoundtrip_ShouldMatchOriginalInput(HttpHeaders headers)
+        {
+            Memory<byte> encoding = HPackEncode(headers);
+            HttpHeaders decodedHeaders = HPackDecode(encoding);
+
+            CompareHttpHeaders(headers, decodedHeaders);
+        }
+
+        private static HttpHeaders BuildHttpHeaders(params (string key, string[] values)[] seedValues)
+        {
+            var headers = new HttpRequestHeaders();
+
+            foreach ((string key, string[] value) header in seedValues)
+            {
+                headers.Add(header.key, header.value);
+            }
+
+            return headers;
+        }
+
+        private static void CompareHttpHeaders(HttpHeaders expected, HttpHeaders actual)
+        {
+            Assert.Equal(expected.Count(), actual.Count());
+            Assert.All(expected.Zip(actual), x => CompareHttpHeader(x.First, x.Second));
+
+            void CompareHttpHeader(KeyValuePair<string, IEnumerable<string>> expected, KeyValuePair<string, IEnumerable<string>> actual)
```

Comment:
The Assert.All above is already using a lambda... why have it then call a local function that's only being used once rather than just inlining this into the lambda?

Reply:
Done.